pkg/synopsysctl/blackduck_migrate.go

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright (C) 2019 Synopsys, Inc.
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package synopsysctl
+
+import (
+	"fmt"
+	v1 "github.com/blackducksoftware/synopsys-operator/pkg/api/blackduck/v1"
+	"github.com/blackducksoftware/synopsys-operator/pkg/blackduck"
+	"github.com/blackducksoftware/synopsys-operator/pkg/util"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+)
+
+func migrate(bd *v1.Blackduck, operatorNamespace string) error {
+	// TODO ensure operator is installed and running a recent version that doesn't require additional migration
+
+	log.Info("Stopping Operator")
+	soOperatorDeploy, err := util.GetDeployment(kubeClient, operatorNamespace, "synopsys-operator")
+	if err != nil {
+		return err
+	}
+
+	soOperatorDeploy, err = util.PatchDeploymentForReplicas(kubeClient, soOperatorDeploy, util.IntToInt32(0))
+	if err != nil {
+		return err
+	}
+
+	// Generate Helm configuration
+	helmValuesMap, err := BlackduckV1ToHelm(bd, operatorNamespace)
+	if err != nil {
+		return err
+	}
+
+	log.Info("Deleting Blackduck resources")
+	// TODO wait for resources to be deleted
+	if err := deleteComponents(bd); err != nil {
+		return err
+	}
+
+	log.Info("Upgrading Blackduck using Helm")
+	chartRepo := "https://artifactory.internal.synopsys.com/artifactory/bds-hub-helm-snapshot-local/blackduck/blackduck-2020.4.0.tgz"
```

Comment:
You might already be planning to fix all the chartRepo variables but just in case this is hard coded here :)